### PR TITLE
Add clipboard support

### LINKED ISSUE #303: Possible to copy paste into gamescope windows?
It seems that gamescope windows have isolated copy and paste, i was wondering if theres any way i can copy from outside gamescope into it?, right now i cant copy my password for example into the gamescope window. Ty for your help. Im running on Sway if it matters

### PR DESCRIPTION
**Pull request by ForTheReallys**
Tested with `gamescope gvim`.  This PR adds support for copying/pasting utf-8 text for both the clipboard and the primary selection.  In the event that SDL_CLIPBOARDUPDATE doesn't get triggered (for example, when running gamescope as XWayland under sway), I've implemented the keybinding `Super + C` to sync both the clipboard and primary selection with gamescope.  As of now, this PR only supports Xwayland application windows under gamescope.  I don't have much experience with Xlib, so any feedback is appreciated.

Closes #303

### CODE REVIEW

File: README.md
Comment (Joshua-Ashton) [inline README.md:43]: C

**Comment by emersion** [comment]
>  In the event that SDL_CLIPBOARDUPDATE doesn't get triggered (for example, when running gamescope as XWayland under sway), I've implemented the keybinding Super + C to sync both the clipboard and primary selection with gamescope.

That sounds like a bug somewhere. Sway should update the X11 clipboard when an X11 window is focused.

**Comment by ForTheReallys** [comment]
@emersion Just found out it's a bug in SDL.  Looks SDL doesn't send CLIPBOARDUPDATE if the videodriver is x11.

**Comment by Joshua-Ashton** [comment]
We can just fix it in SDL maybe?

**Comment by ForTheReallys** [comment]
Works for me.  I'll look into that.  In the meantime, is there anything stopping us from merging this PR?

**Comment by ForTheReallys** [comment]
SDL PR: https://github.com/libsdl-org/SDL/pull/7951

**Comment by ForTheReallys** [comment]
SDL PR: https://github.com/libsdl-org/SDL/pull/7951

**Comment by ForTheReallys** [comment]
The fix has been merged into SDL2.  Let me know if there's anything else you need from me

**Comment by ForTheReallys** [comment]
@emersion @Joshua-Ashton Bump

**Comment by Joshua-Ashton** [comment]
On Holiday until Wednesday

**Comment by ForTheReallys** [comment]
Any updates on this @Joshua-Ashton?

**Comment by ForTheReallys** [comment]
> With the SDL bug fixed, do we really need the keyboard shortcut to synchronize?

Really your call. I don't mind either way. I'm not super familiar with distro release schedules, but the only reason I can think of to keep it around is if this change gets picked up by distros before the SDL fix. Again, your call. Let me know and I'll do what I need to do

**Comment by ForTheReallys** [comment]
@emersion I went ahead and removed the keybind now that SDL has been fixed to properly report clipboard updates.  I also added support for clipboard synchronization across xwayland instances in embedded mode.  I was going to wait until this PR got merged, but it was a pretty simple change, so I just included it in this PR.  That commit should also fix #916.

File: src/steamcompmgr.cpp
Comment (Joshua-Ashton) [inline src/steamcompmgr.cpp:4625]: Should call this like the other things that loop with `gamescope_xwayland_server_t` and also avoid updating the atom for the server that's calling it as we don't want to end up in a race with buffered messages on the same server.

**Review by Joshua-Ashton — CHANGES_REQUESTED** [review]

File: src/steamcompmgr.cpp
Comment (ForTheReallys) [reply src/steamcompmgr.cpp:4625]: Can you elaborate on what you mean by "like the other things that loop with `gamescope_xwayland_server_t`?"

Comment (Joshua-Ashton) [reply src/steamcompmgr.cpp:4625]:
```
		gamescope_xwayland_server_t *server = NULL;
		for (size_t i = 0; (server = wlserver_get_xwayland_server(i)); i++)
```

Comment (ForTheReallys) [reply src/steamcompmgr.cpp]: As it turns out, we shouldn't actually need to update the atom at all with this call in any of the instances, since that gets done when we receive a SelectionRequest.  All we really want to do here is become the selection owner.  Again, I'm not all that familiar with Xlib, so please let me know if I'm missing something

**Comment by ForTheReallys** [comment]
Bump @Joshua-Ashton

**Comment by ForTheReallys** [comment]
Bump @Joshua-Ashton

**Comment by Joshua-Ashton** [comment]
Thought I merged this already, oops!

**Comment by ForTheReallys** [comment]
No worries.  Thank you!

**Comment by Preposterone** [comment]
This PR has been merged, however, I don't see this functionality neither in readme of current codebase, nor in help message of a built-from source executable.

Has this been reverted for some reason? If not, what are the steps to enable basic clipboard functionality?

@ForTheReallys @Joshua-Ashton

**Comment by ForTheReallys** [comment]
@Preposterone the easiest way to get this working is to set `SDL_VIDEODRIVER=wayland`  (assuming you're on Wayland). For it to work with X11 or XWayland, you'll need to have SDL 2.30.0.

**Comment by MisterrrX** [comment]
> @Preposterone the easiest way to get this working is to set `SDL_VIDEODRIVER=wayland` (assuming you're on Wayland). For it to work with X11 or XWayland, you'll need to have SDL 2.30.0.

Unfortunately, this doesn't work for me on Arch and Hyprland. Has anyone got this to work?

**Comment by mikeyjoel** [comment]
I play FFXVI often and need to login using my SquareEnix Mogstation provided credentials. I can't copy and paste from 1Password onto it. Tried some a different app, doesnt work. Disabling gamescope allows for clipboard functionality. Using Tumbleweed with Steam installed from the games repo via zypper natively.

**Comment by Artisme** [comment]
Hi, is there a way to make this work on Arch and Hyprland

**Comment by RheaAyase** [comment]
This is pretty old, how come that it's still not functional? I highly doubt that my version is "that old" :D
No worky on 15.2.1 and a bit newer one I didn't check version number for when I tried...

**Pull request merged**
